### ISSUE TITLE
Add "Custom" propagation method or Event in `getSupportedSiteIds` for Matrix fields as well

### ISSUE THREAD

**Issue by Anubarak** [craftcms/cms#7610]
### Description

**TL/DR**: we just need a "custom" propagation method, like most other fields have. So we like to decide what sites a matrix field should be propagated to. That could be made via an Event in `craft\services\Matrix::getSupportedSiteIds` when you add the `field` as an optinal parameter.

**Long Story**:

Why do we need this?
We have a client with an incredible complex localization structure like the following
* en
  * en-US
  * en-AU 
* fr-Fr
  * fr-CA
  * fr-BE
* de-De
  * de-Ch  
  * .....
* .... 

We implemented a lightswitch in all entries `hasOwnContent` if that lightswitch is `false`, the content will be populated from it's "parent" culture/site. 
When they create an new entry they don't want to necessarily write that one in all 55 languages in one day, so even eg `fr-Fr` should pull it's content from `en` (just their wish, I'm not responsible for their ideas). That's why the default propagation methods won't work and we need a custom logic to adjust the propagation based on the field values of the entry. 

For example: saving an entry in `en` should update all other sites (all the way down in the structure) unless `hasOwnContent` is `true`. Saving `de-DE` should propagate to all `de-XX` sites unless `hasOwnContent` is `true` and so on.

**Comment by brandonkelly** [comment]
Just added this for 3.7.

To update to the 3.7 branch locally, change your `craftcms/cms` requirement in composer.json to `"3.7-dev"`, and run `composer update`.

**Comment by Anubarak** [comment]
Awesome as always, thank you 👍 
I really appreciate your super fast reactions
